### PR TITLE
feat(mobile): enhance download operations

### PR DESCRIPTION
**Pull request by alextran1502**
This PR uses the `background_downloader` package to enhance the download operation. It will now show the progress of the download and allow the user to cancel it.

In the near future, this package will also be used to upload assets.

### Download progress and cancel operation
image

### CODE REVIEW

**Comment by bo0tzz** [comment]
Closes #10100

File: mobile/lib/main.dart [hunk 1]
```diff
@@ -87,6 +89,17 @@ Future<void> initApp() async {
   };
 
   initializeTimeZones();
+
+  FileDownloader().configureNotification(
+    running: const TaskNotification('Downloading media', 'file: {filename}'),
+    complete: const TaskNotification('Download finished', 'file: {filename}'),
```
Comment (jrasm91) [inline mobile/lib/main.dart:95]: I assume these should be translated.

File: mobile/lib/pages/common/download_panel.dart [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        case TaskStatus.canceled:
+          return 'Download canceled';
+        case TaskStatus.paused:
+          return 'Download paused';
+        case TaskStatus.enqueued:
+          return 'Download enqueued';
+        case TaskStatus.notFound:
+          return 'Download not found';
+        case TaskStatus.waitingToRetry:
+          return 'Download waiting to retry';
```
Comment (jrasm91) [inline mobile/lib/pages/common/download_panel.dart:93]: This as well

File: mobile/lib/services/download.service.dart [hunk 3]
```diff
@@ -0,0 +1,183 @@
+import 'dart:io';
+
+import 'package:background_downloader/background_downloader.dart';
```
Comment (jrasm91) [inline mobile/lib/services/download.service.dart:3]: Can you rework this to use the repository pattern? With it you should not import this dependency here at all. We could make a new DownloadRepository with methods to download, and add progress/state callbacks.

File: mobile/lib/services/download.service.dart [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        ),
+      );
+    } else {
+      await FileDownloader().enqueue(
+        _buildDownloadTask(
+          asset.remoteId!,
+          asset.fileName,
+          group: asset.isImage ? downloadGroupImage : downloadGroupVideo,
+        ),
+      );
```
Comment (jrasm91) [inline mobile/lib/services/download.service.dart:159]: This should just be a single repo.addTask or repo.download method here.

File: mobile/pubspec.yaml [hunk 5]
```diff
@@ -32,7 +32,7 @@ dependencies:
   flutter_svg: ^2.0.9
   package_info_plus: ^8.0.1
   url_launcher: ^6.2.4
-  http: ^0.13.6
+  http: ^1.1.0
```
Comment (jrasm91) [inline mobile/pubspec.yaml:35]: Why is this updated? Can we bump the dependencies in a separate PR? Presumably it should be tested along with other functionality unrelated to file downloads.

**Review by jrasm91 — APPROVED** [review]

File: mobile/pubspec.yaml (hunk 5 above)
Comment (alextran1502) [reply mobile/pubspec.yaml:35]: It is the required version by the background_downloader package
